scripts/build_dists_archive.sh

```diff
@@ -33,7 +34,8 @@ if [ -f "$envFile" ]; then
 fi
 
 if [ -f "Cargo.toml" ]; then
-  if [ "$1" == "clean" ]; then
+  if [ "$1" == "clean" ] || [ "$(uname)" == "Darwin" ]; then
```

Comment:
The OR part is going to break this script, as it's going to shift the command line options, if it's OSX or if the first options is ```clean```. Unless you rework this totally, I would suggest removing the OR Darwin bit. This should be up for the build to know.

Reply:
I haven't found a way yet to set env variables using cargo for the build itself without going further into nightly rust. 

For mac we run into the problem with open_ssl not being signed if it is dynamically linked, so we need to ensure that this is statically linked here.

This is meant to catch the case of human error when building.

Assume clean always by default?

Reply:
This script is only going to be run by Distro Builders, so I think that forcing clean build for OSX builder every time is wasteful. This only needs to be the case on first build after switching to ```statically linked```? These builds in GHA would always be clean builds, but if you testing locally, doubling your build time seems wrong.  
  
If you wish to always do clean builds, remove the options as an override. I don't have an issue with that, but having this script break for OSX users, because of the use of ```shift``` in the if block, that is just going to be poor.

Reply:
Removed the shift and clean argument.